framework/include/kernels/ADMaterialPropertyValue.h

```diff
@@ -0,0 +1,26 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "ADKernelValue.h"
+
+class ADMaterialPropertyValue : public ADKernelValue
```

Comment:
Any reason you're not combining the AD and non-AD versions?

Reply:
The inheritance of `ADKernelValue` / `KernelValue`.  I thought a `GenericKernelValue` would be needed to combine bother versions

Reply:
No, check out https://github.com/idaholab/moose/blob/06bc1f73f0d4dfb3d01ce9ba77d6bb97346e07eb/modules/tensor_mechanics/include/bcs/Torque.h#L17-L25 for example

Reply:
Nice trick!